engine/app/components/citizens_advice_components/targeted_content.rb

```diff
@@ -0,0 +1,74 @@
+# frozen_string_literal: true
+
+module CitizensAdviceComponents
+  class TargetedContent < Base
+    attr_reader :id, :type, :heading_level
+
+    # rubocop:disable Metrics/MethodLength, Metrics/ParameterLists
+    def initialize(
+      title:,
+      id:,
+      type: :public,
+      heading_level: nil,
+      show_close_button: nil,
+      is_toggleable: nil
+    )
+      super
+      @title = title
+      @id = id
+      @type = fetch_or_fallback(
+        allowed_values: %i[public adviser],
+        given_value: type,
+        fallback: :public
+      )
+      @heading_level = (heading_level || 2).to_i.clamp(2, 6)
```

Comment:
you could reduce the cyclomatic score by abstracting this out to a method below. Maybe `detect_heading_level` as a name?